### PR TITLE
Fixes #16001: Add logging contexts

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/16001

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR rebased

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR rebased

File: relay/sources/relayd/src/api/shared_folder.rs [hunk 1]
```diff
@@ -53,46 +51,33 @@ fn default_hash() -> String {
...
-                )
-            } else {
-                reply::with_status(
-                    format!(
-                        "{:#} exists but its hash is which is different",
-                        file.display()
-                    ),
-                    StatusCode::OK,
-                )
+    futures::future::result(HashType::from_str(&params.hash_type)).and_then(move |hash_type| {
```
Comment (ncharles) [inline relay/sources/relayd/src/api/shared_folder.rs]: there's no checking anymore on parameters, is it expected ?

**Comment by amousset** [comment]
PR updated with a new commit

File: relay/sources/relayd/src/api/shared_folder.rs (hunk 1 above)
Comment (amousset) [reply relay/sources/relayd/src/api/shared_folder.rs]: We lost a bit of log precision indeed! I'm fixing this.

**Pull request merged**